pkg/providers/aws/provider_redis.go

```diff
@@ -166,6 +179,84 @@ func (p *AWSRedisProvider) createElasticacheCluster(ctx context.Context, r *v1al
 	}}, croType.StatusMessage(fmt.Sprintf("creation successful, aws elasticache status is %s", *foundCache.Status)), nil
 }
 
+// Add Tags to AWS Elasticache
+func (p *AWSRedisProvider) TagElasticache(ctx context.Context, r *v1alpha1.Redis, stratCfg StrategyConfig, cacheClusterId string, preferredAvailabilityZone string) (types.StatusMessage, error) {
+	cacheSvcTag := elasticache.New(session.Must(session.NewSession(&aws.Config{Region: aws.String(stratCfg.Region)})))
+	cacheRegion := preferredAvailabilityZone
+	cacheRegion = cacheRegion[:(len(preferredAvailabilityZone))-1]
```

Comment:
i'm a bit out of the loop here, i'm not sure i understand the redeclaration here, what does `cacheRegion` look like at each of these declarations?

Reply:
Basically preferredAvailabilityZone returns the sub zone like eu-west1b, by dropping the last character we get the main zone eu-west1,which is what the api requires, is a bit hacky

Reply:
Ah ok, can we add that in as a comment in the code? We could hardcode a lookup table. But I believe this approach will work, makes sense.